packages/gatsby-source-filesystem/src/utils.js

```diff
@@ -39,3 +39,22 @@ export function getRemoteFileExtension(url) {
 export function getRemoteFileName(url) {
   return getParsedPath(url).name
 }
+
+/**
+ * slash
+ * --
+ * Convert Windows backslash paths to slash paths: foo\\bar ➔ foo/bar
+ *
+ *
+ * @param  {String}          path
+ * @return {String}          slashed path
+ */
+export function slash(path) {
+  const isExtendedLengthPath = /^\\\\\?\\/.test(path)
```

Comment:
Windows supports a special syntax for pathnames longer than 255 characters (you can read more [here](https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#maximum-path-length-limitation)).

`slash` should not modify pathnames in this special syntax because they are only valid on Windows machines and thus must use forward slashes. You should add a test asserting that this is the case, like

```
it(`does not modify extended length paths`, () => {
    const extended = `\\\\?\\some\\path`
    expect(slash(extended)).toBe(extended)
})
```

Reply:
oh,that what you mean...

Reply:
I have already updated.